src/core/positioning/positioningsource.h

```diff
@@ -0,0 +1,229 @@
+     */
+    int averagedPositionCount() const { return static_cast<int>( mCollectedPositionInformations.size() ); }
+
+    /**
+     * Returns the current elevation correction mode.
+     * \note Some modes depends on device capabilities.
+     */
+    ElevationCorrectionMode elevationCorrectionMode() const { return mElevationCorrectionMode; }
+
+    /**
+     * Sets the current elevation correction mode.
+     * \note Some modes depends on device capabilities.
+     */
+    void setElevationCorrectionMode( ElevationCorrectionMode elevationCorrectionMode );
+
+    /**
+     * Sets the GNSS device antenna height. This should be the pole height + sensore phase height.
+     * \note When IMU is active this value is ignored as the device does the correction internally.
+    **/
+    double antennaHeight() const { return mAntennaHeight; }
+
+    /**
+     * Returns the GNSS device antenna height. This should be the pole height + sensore phase height.
+     * \note When IMU is active this value is ignored as the device does the correction internally.
+    **/
+    void setAntennaHeight( double antennaHeight );
+
+    /**
+     * Returns the current device orientation
+     */
+    double orientation() const { return mOrientation; }
+
+    /**
+     * Returns whether GNSS devices will log their incoming position stream into a logfile.
+     * \note Requires a device type with logging capability
+     */
+    bool logging() const { return mLogging; }
+
+    /**
+     * Sets whether GNSS devices will log their incoming position stream into a logfile.
+     * \note Requires a device type with logging capability
+     */
+    void setLogging( bool logging );
+
+  signals:
+    void activeChanged();
+    void validChanged();
+    void deviceIdChanged();
+    void deviceChanged();
+    void deviceLastErrorChanged();
+    void deviceSocketStateChanged();
+    void deviceSocketStateStringChanged();
+    void positionInformationChanged();
+    void averagedPositionChanged();
+    void averagedPositionCountChanged();
+    void elevationCorrectionModeChanged();
+    void antennaHeightChanged();
+    void orientationChanged();
+    void loggingChanged();
+
+  public slots:
+
+    void triggerConnectDevice();
+    void triggerDisconnectDevice();
+
+  private slots:
+
+    void lastGnssPositionInformationChanged( const GnssPositionInformation &lastGnssPositionInformation );
+    void processCompassReading();
+
+  private:
+    void setupDevice();
+
+    bool mActive = false;
+
+    QString mDeviceId;
+    bool mValid = false;
+
+    GnssPositionInformation mPositionInformation;
+    QList<GnssPositionInformation> mCollectedPositionInformations;
+
+    bool mAveragedPosition = false;
+
+    ElevationCorrectionMode mElevationCorrectionMode = ElevationCorrectionMode::None;
+    double mAntennaHeight = 0.0;
+
+    bool mLogging = false;
+
+    AbstractGnssReceiver *mReceiver = nullptr;
+
+    QCompass mCompass;
+    QTimer mCompassTimer;
```

Comment:
Was there a reason to add a timer instead of using `mCompass.readingChanged()`?
If yes, we should leave this as a comment.